src/components/NewMessageForm/NewMessageForm.vue

```diff
@@ -285,7 +285,9 @@ export default {
 			for (const index in shareableFiles) {
 				// The pat of the file to share to the conversation
 				const path = this.$store.getters.getAttachmentFolder() + '/' + shareableFiles[index].name
-				shareFile(path, token)
+				// Mark current file as shared
+				this.$store.dispatch('markFileAsShared', { token, index })
```

Comment:
this has to happen for all files before the first `await shareFile`
Or you have to check the state within the loop (if it even updates), to make sure no other execution is sharing it between you sharing item1 and item2.